.eslintrc.js

```diff
@@ -11,6 +11,46 @@ module.exports = {
     'src/govuk/vendor/polyfills/**/*'
   ],
   overrides: [
+    {
+      extends: 'plugin:jsdoc/recommended',
+      files: ['**/*.{cjs,js,mjs}'],
+      plugins: [
+        'jsdoc'
+      ],
+      rules: {
+        // JSDoc blocks are optional
+        'jsdoc/require-jsdoc': 'off',
+        'jsdoc/require-param-description': 'off',
+        'jsdoc/require-param': 'off',
+
+        // Require hyphens before param description
+        // Aligns with TSDoc style: https://tsdoc.org/pages/tags/param/
+        'jsdoc/require-hyphen-before-param-description': 'warn',
+
+        // Check for valid formatting
+        'jsdoc/check-line-alignment': 'warn',
+        'jsdoc/check-syntax': 'error',
+
+        // Add unknown @jest-environment tag name
+        'jsdoc/check-tag-names': [
+          'warn', {
+            definedTags: ['jest-environment']
+          }
+        ],
+
+        // Add missing .querySelectorAll() type
+        'jsdoc/no-undefined-types': [
+          'error', {
+            definedTypes: ['NodeListOf']
+          }
+        ]
+      },
+      settings: {
+        jsdoc: {
+          mode: 'typescript'
```

Comment:
Ah, I was reviewing commit by commit so didn't see that. I think it might be worth capturing as a comment in the config file? Perhaps:

```suggestion
         // Allows us to re-use declarations that exist in our dependencies
          mode: 'typescript'
```